src/main/java/com/connexta/store/service/impl/StoreServiceImpl.java

```diff
@@ -216,4 +216,39 @@ public void quarantine(
       adaptor.delete(datasetId);
     }
   }
+
+  @Override
+  public void unstage(
+      @Pattern(regexp = "^[0-9a-zA-Z]+$") @Size(min = 36, max = 36) final String datasetId)
+      throws DatasetNotFoundException, QuarantineException {
+    fileStorageAdaptor.updateStatus(datasetId, STORED);
+    metacardStorageAdaptor.delete(datasetId);
+    log.info("Unstaged file and deleted staged metacard for datasetId={{}}", datasetId);
+  }
+
+  private void verifyStoreStatus(String datasetId, StorageAdaptor adaptor) {
+    String status = adaptor.getStatus(datasetId);
+    // TODO: Should the exception that is returned depend on what the status is (Staged vs
+    // Quarantined)?
+    if (status == null || !status.equals(STORED)) {
+      log.info("Retrieval failed. Status for datasetId={{}} is not \"stored\".", datasetId);
+      throw new RetrieveException(
+          String.format("Dataset for Id={%s} has not been stored", datasetId));
+    }
+  }
+
+  @NotNull
+  private Resource validateMetadataResource(ResponseEntity<Resource> metadataResource) {
+    if (metadataResource == null) {
```

Comment:
❓ I'm not sure if/how we want to handle this.  Should the Service know anything about HTTP status codes?  What if the Service was used outside of an endpoint?  Wouldn't it be weird to use HTTP status codes in that situation?  Thoughts?

Reply:
This was just refactored from `addMetadata` to reduce complexity. @peterhuffer could chime in. My two cents are to throw a custom exception.

Reply:
@figliold no we shouldn't be. These interfaces should define exceptions that are thrown and caught by an exception handler which will translate them to the appropriate HTTP response. Our interfaces were throwing these HTTP response exceptions though already so this is something we need to address later.